samples/net/sockets/tls_configurations/README.rst

```diff
@@ -0,0 +1,98 @@
+
+Overview
+********
+
+This example showcases various minimal Mbed TLS configurations to properly set up
+a socket-tls client connection to a server. However since testing all possible
+ciphersuites of TLS 1.2 and 1.3 might take a considerable amount of test time
+and increase the size of :file:`sample.yaml` file for no valid reason, only some
+relevant case is proposed:
+
+* TLS 1.2
+
+   * ``sample.net.sockets.tls12.rsa_kex``
+      using RSA for key exchange and certificate validation
+   * ``sample.net.sockets.tls12.ec_kex``
+      relying elliptic curves algorithms, i.e. ECDHE for key exchange and ECDSA
+      for certificate validation
+
+* TLS 1.3
+
+   * ``sample.net.sockets.tls13.ec_kex.no_tickets``
+      using ECDHE key exchange and EDCSA certificate, but without session tickets
+   * ``sample.net.sockets.tls13.ec_kex``
+      using ECDHE key exchange and EDCSA certificate with session tickets
+   * ``sample.net.sockets.tls13.psk_kex``
+      using pure PSK key exchange without any certificate.
+
+Supported platform
+******************
+
+Only ``native_sim`` or ``native_sim/native/64`` are supported by this sample. The
+reason is that:
+
+* on one hand TLS sockets are basically a software component so there is no need
+  to test (i.e. depend) on some specific HW;
+* on the other one it simplifies the code (no TCP connection to handle for
+  example) and test (it can be easily tested against an OpenSSL server running
+  on the same PC).
+
+For this reason native offloaded sockets are used in this sample.
+
+.. _how_to_build:
+
+How to build
+************
+
+Here's an example on how to build the ``sample.net.sockets.tls13.ec_kex``
+configuration:
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/net/sockets/tls_configurations
+   :tool: west
+   :goals: build
+   :host-os: unix
+   :board: native_sim/native/64
+   :west-args: -T sample.net.sockets.tls13.ec_kex
+   :compact:
+
+All other configurations can be build similarly by just replacing the test item
+with those listed in the :file:`sample.yaml`.
+
+How to test
+***********
+
+OpenSSL server can easily be launched through the script
+:file:`scripts/start-server.sh`. It requires 1 input parameter which is used to
+select the proper configuration to be used:
+
+* ``1.2-rsa``
+* ``1.2-ec``
+* ``1.3-ec-no_tickets``
+* ``1.3-ec``
+* ``1.3-psk``
+
+For example:
+
+.. code-block:: console
+
+   ./scripts/start_server.sh 1.3-ec
+
+will start a server instance which is compatible with the Zephyr application
+built in :ref:`how_to_build` section.
+Of course each Zephyr configuration listed in :file:`sample.yaml` requires
+the proper server counterpart in order to work successfully.
+
+
+Regenerate certificates and keys [optional]
+*******************************************
+
+If desired EC/RSA certificates and keys used in the sample can be re-generated
+using the :file:`scripts/create-certs.sh` scripts. However this is totally
+optional and not required to run the sample.
```

Comment:
It's not strictly necessary but I thought it could be useful to show how to re-generate certificate and keys using a standard Linux tool. If you think this is not important I can remove the script and this section from the readme file.